src/main/scala/org/apache/spark/sql/connector/kinesis/ShardSyncer.scala

```diff
@@ -263,6 +263,19 @@ object ShardSyncer extends Logging {
             new ShardInfo(shardId, initialPosition.shardPosition(shardId)))
         }
     }
+
+    closedShards(latestShards).foreach {
```

Comment:
I think that logic needs to change. Instead of iterating over `latestShards` to get collection of open shards and closed shards, and then iterating separately over each collection, it should be a single pass over `latestShards` and depending on whether shard is open or closed, adding different `ShardsInfo`.

Reply:
I changed the code to do a single pass over latestShards though it looked cleaner before.